chapters/en/slides/module5/module5_04.Rmd

```diff
@@ -0,0 +1,348 @@
+    x='Horsepower',
+    y='Miles_per_Gallon',
+    color=alt.Color('Origin', title=None, scale=alt.Scale(range=colors)))
+```
+
+Notes:
+Most of the time it is a good idea to stick to the predefined color scheme
+because of the advantages mentioned in the previous slide.
+
+However,
+sometimes the categories we are representing might have a color already associated with them,
+such as political parties or sports team.
+
+In these cases,
+it is often better to design a custom color scheme
+using the colors naturally associated with each category.
+
+In Altair we can create a custom color scheme by passing a list of colors 
+to the `range` parameter of `alt.Scale`.
+
+You can specify colors either by their HTML/CSS name (such as 'coral')
+or their hex code (such as '#4682b4').
+[All HTML/CSS color names can be found in the image in this post](https://stackoverflow.com/a/37232760/2166823).
+
+Hex codes are defines over the range `#000000` for black ("zero color")
+to `#ffffff` for white ("full color")
+and [are easiest chosen via color pickers such as this one](https://colorpicker.me/).
+
+---
+
+## Don't use more than 5-8 distinct hues for categories
+
+<center><img src=/module5/hue-guidelines.png></img></center>
+
+Notes:
+Although many of the categorical color schemes contain ten or more hues,
+if is often not a good idea to use all that many.
+
+The guidelines on what is too many hues differ between different sources
+and also depends on your use case.
+A good rule of thumb is that when you get to around five different hues,
+you should really consider if this is the best way to represent your data
+or if you could split it up into multiple visualizations instead.
+
+If the data is neatly organized in well-separated clusters,
+it is possible that you could visualize more than five colors
+(maybe even ten) effectively,
+but in data where the datapoints are more mixed,
+you will rarely, if ever, be able to go this high.
+
+---
+
+## Too many hues are impossible to distinguish
+
+```{python}
+alt.Chart(cars).mark_point(size=70, filled=True).encode(
+    x='Horsepower',
+    y='Miles_per_Gallon',
+    color=alt.Color('Name', title=None))
+```
+
+Notes:
+This is an example of what happens when too many categorical hues are used.
+In this plot,
+we can not separate the colors
+even if we spent a considerable amount of effort studying the chart.
+
+As we saw on the last slide,
+categorical color schemes have a limited amount of hues
+so in addition of it being hard to differentiate this many hues in general,
+it become practically impossible in cases
+where there are more categories than coor hues,
+since the color scheme starts repeating as in the chart.
+
+A better approach here would have been to label specific points of interest directly
+and keeping the rest as either a single color or using a categorical variable with fewer values, e.g. by grouping the cars into brands rather than their full model name.
```

Comment:
Can we add a slide showing this better way of visualizing this data? I think the learners would really benefit from seeing this, over having a verbal description.